lib/improver/utilities/cube_checker.py

```diff
@@ -52,3 +56,136 @@ def check_for_x_and_y_axes(cube):
             msg = ("The cube does not contain the expected {}"
                    "coordinates.".format(axis))
             raise ValueError(msg)
+
+
+def check_cube_coordinates(cube, new_cube, exception_coordinates=None):
+    """
+    Find and promote to dimension coordinates any scalar coordinates in
+    new_cube that were originally dimension coordinates in the progenitor
+    cube. If coordinate is in new_cube that is not in the old cube, keep
+    coordinate in its current position.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube
+        The input cube provided to nbhood.
```

Comment:
This is now moved to a more general place so we should remove all references to nbhood and neighbourhooding.